File: C++SrcFiles/Cathulubullet.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cmath>
#include <algorithm>

using namespace std;

class Cathulubullet {
private:
    GameObject* attackPrefab;
    Transform* attackPosition;
    float attackTimer;

public:
    Cathulubullet(GameObject* attackPrefab, Transform* attackPosition) {
        this->attackPrefab = attackPrefab;
        this->attackPosition = attackPosition;
        attackTimer = 0.0f;
    }

    void Update() {
        attackTimer += Time.deltaTime;
        if (attackTimer > 2.0f) {
            attackTimer = 0.0f;
            Shoot();
        }
    }

    void Shoot() {
        Instantiate(attackPrefab, attackPosition->position, Quaternion::identity);
    }
};

File: C#SrcFiles/Cathulubullet.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Cathulubullet : MonoBehaviour
{
    // the prefab of the Cathuluattack png;
    public GameObject attack;
    //position of the attack;
    public Transform attackposition;

    // Intervals between attacks;
    private float attacktimer;
    void Update()
    {
        // counting till 3 seconds and then shooting the reaper bullet.
        attacktimer += Time.deltaTime;
        if (attacktimer > 2)
        {
            attacktimer = 0;
            shoot();
        }
    }


    private void shoot()
    {
        Instantiate(attack, attackposition.position, Quaternion.identity);
    }
}
